frontend/src/metabase/visualizations/lib/settings/column.js

```diff
@@ -502,52 +502,94 @@ export const getTitleForColumn = (column, series, settings) => {
 
 export const buildTableColumnSettings = ({
   getIsColumnVisible = col => col.visibility_type !== "details-only",
-} = {}) => ({
-  // NOTE: table column settings may be identified by fieldRef (possible not normalized) or column name:
-  //   { name: "COLUMN_NAME", enabled: true }
-  //   { fieldRef: ["field", 2, {"source-field": 1}], enabled: true }
-  "table.columns": {
-    section: t`Columns`,
-    title: t`Columns`,
-    widget: ChartSettingTableColumns,
-    getHidden: (series, vizSettings) => vizSettings["table.pivot"],
-    isValid: ([{ card, data }]) => {
-      const columns = card.visualization_settings["table.columns"];
-      const enabledColumns = columns.filter(column => column.enabled);
-      // If "table.columns" happened to be an empty array,
-      // it will be treated as "all columns are hidden",
-      // This check ensures it's not empty,
-      // otherwise it will be overwritten by `getDefault` below
-      return (
-        card.visualization_settings["table.columns"].length !== 0 &&
-        _.all(
-          enabledColumns,
-          columnSetting =>
-            findColumnIndexForColumnSetting(data.cols, columnSetting) >= 0,
-        )
-      );
-    },
-    getDefault: ([
+} = {}) => {
+  const isValid = ([{ card, data }], settings) => {
```

Comment:
I extracted isValid/getDefault/getValue as const outside of the big returned object because I needed to call getDefault/isValid from getValue